app/lib/view_model/logic/favorite/IFavoriteMealAccess.dart

```diff
@@ -0,0 +1,25 @@
+
+import '../../repository/data_classes/meal/Meal.dart';
+
+/// This class is the interface for the access to the favorite meals data.
+abstract class IFavoriteMealAccess {
+
+  /// This method adds the committed meal to the favorite meals in the database.
+  /// @param meal The meal that should be added
+  /// @return The result of the update
+  Future<void> addFavoriteMeal(Meal meal);
+
+  /// This method removes the committed meal from the favorite meals in the database.
+  /// @param meal The meal that should be removed
+  /// @return The result of the update
+  Future<void> removeFavoriteMeal(Meal meal);
+
+  /// This method checks if the committed meal is a favorite meal.
+  /// @param meal The meal that should be checked
+  /// @return ture, if the meal is a favorite meal, false otherwise
+  Future<bool> isFavoriteMeal(Meal meal);
+
+  /// This method returns the favorite meals from the database.
+  /// @return The favorite meals or an error
+  Future<List<String>> getFavoriteMeals();
```

Comment:
I suspect you only return the meals' ids here (maybe an id type would make that clearer).
I find that somewhat interesting as you need the complete meal to add it to the favorites.